src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2880,6 +2880,12 @@ FindReferencedTableColumn(Expr *columnExpression, List *parentQueryList, Query *
 		return;
 	}
 
+	/* this function doesn't support finding the RTEs above this query */
+	if (candidateColumn->varlevelsup > 0)
+	{
+		return;
```

Comment:
I would appreciate a slightly more elaborate comment on what the effect of this is.

Reply:
updated the comment